src/wrappers/OsipiBase.py

```diff
@@ -0,0 +1,210 @@
+            else:
+                # Bounds may not be required but can be optional
+                if self.required_bounds_optional:
+                    print(f"Bounds required: {self.required_bounds} but is optional")
+                else:
+                    print(f"Bounds required: {self.required_bounds} and is not optional")
+                    
+        if hasattr(self, "required_initial_guess") and hasattr(self, "required_initial_guess_optional"):
+            if self.required_initial_guess:
+                print(f"Initial guess required: {self.required_initial_guess}")
+            else:
+                # Bounds may not be required but can be optional
+                if self.required_initial_guess_optional:
+                    print(f"Initial guess required: {self.required_initial_guess} but is optional")
+                else:
+                    print(f"Initial guess required: {self.required_initial_guess} and is not optional")
+
+    def osipi_accepted_dimensions(self):
+        """The array of accepted dimensions
+        e.g.
+        (1D,   2D,   3D,    4D,    5D,    6D)
+        (True, True, False, False, False, False)
+        """
+        
+        return (False,) * 6
+
+    def osipi_accepts_dimension(self, dim):
+        """Query if the selection dimension is fittable"""
+        
+        accepted = self.accepted_dimensions()
+        if dim < 0 or dim > len(accepted):
+            return False
+        return accepted[dim]
+    
+    def osipi_check_required_bvalues(self):
+        """Checks if the input bvalues fulfil the algorithm requirements"""
+        
+        if self.bvalues.size < self.required_bvalues:
+            print("Conformance error: Number of b-values.")
+            return False
+        else: 
+            return True
+        
+    def osipi_check_required_thresholds(self):
+        """Checks if the number of input thresholds fulfil the algorithm requirements"""
+        
+        if (len(self.thresholds) < self.required_thresholds[0]) or (len(self.thresholds) > self.required_thresholds[1]):
+            print("Conformance error: Number of thresholds.")
+            return False
+        else: 
+            return True
+        
+    def osipi_check_required_bounds(self):
+        """Checks if input bounds fulfil the algorithm requirements"""
+        if self.required_bounds is True and self.bounds is None:
+            print("Conformance error: Bounds.")
+            return False
+        else:
+            return True
+
+    def osipi_check_required_initial_guess(self):
+        """Checks if input initial guess fulfil the algorithm requirements"""
+        
+        if self.required_initial_guess is True and self.initial_guess is None:
+            print("Conformance error: Initial guess")
+            return False
+        else:
+            return True
+
+    
+    def osipi_check_required_bvalues():
+        """Minimum number of b-values required"""
+        pass
```

Comment:
The other "checks" return `True` or `False`. Perhaps this template should just return `True`?